pages/task/read/views/activity-log.jsx

```diff
@@ -17,36 +31,84 @@ const getStatusBadge = eventName => {
   return <span className={ className }><Snippet>{ `status.${status}.state` }</Snippet></span>;
 };
 
-const getProfile = (profile, establishmentId) => {
-  const name = getName(profile);
-  return profile.asruUser
-    ? name
-    : <Link page="profile.view" profileId={profile.id} establishmentId={establishmentId} label={name} />;
-};
+const getAuthor = (profile, task) => {
+  const name = `${profile.firstName} ${profile.lastName}`;
+  const role = getRole(profile, task);
 
-const ActivityLog = ({ task, ExtraMeta }) => {
-  if (!task.activityLog) {
-    return null;
-  }
+  return (
+    <p>
+      { role && <span className="role">{role}: </span> }
+      {
+        profile.asruUser
+          ? name
+          : <Link page="profile.view" profileId={profile.id} establishmentId={task.data.establishmentId} label={name} />
+      }
+    </p>
+  );
+};
 
+const LogItem = ({ log, task, ExtraMeta }) => {
   return (
-    <Fragment>
-      <h2><Snippet>sticky-nav.activity</Snippet></h2>
-      <ul className="task-activity">
-        { task.activityLog.map(log => (
-          <li key={log.id}>
-            {getStatusBadge(log.eventName)}
-            <p>{getProfile(log.changedBy, task.data.establishmentId)}</p>
-            <p className="comment">{log.comment}</p>
-            {
-              ExtraMeta && <p><ExtraMeta item={log} task={task} /></p>
-            }
-            <p>{format(log.createdAt, dateFormat.datetime)}</p>
-          </li>
-        ))}
-      </ul>
-    </Fragment>
+    <li key={log.id}>
+      <span className="date">{format(log.createdAt, dateFormat.datetime)}</span>
+      {getStatusBadge(log.eventName)}
+      {getAuthor(log.changedBy, task)}
+      <p className="comment">{log.comment}</p>
+      {
+        ExtraMeta && <p><ExtraMeta item={log} task={task} /></p>
+      }
+    </li>
   );
 };
 
+class ActivityLog extends Component {
+  constructor(props) {
+    super(props);
+
+    this.task = props.task;
+    this.ExtraMeta = props.ExtraMeta;
+    this.latestActivity = props.task.activityLog.shift();
```

Comment:
I don't think doing this work in the constructor is correct. If things are props they should just be used directly as props in `render`, and any work to modify them should be done at render-time.

Performing the work at a constructor level means that if the props are mutated then the component will not re-render.

Reply:
The issue here is the `this.latestActivity = props.task.activityLog.shift();` - if you do that in the render then every time you open and close the activity you lose the first entry until there's nothing left.

Reply:
Then don't use `.shift()` - take the first element by doing `activityLog[0]`, and when rendering the rest check `index > 0`.